sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -400,6 +402,22 @@ public SeekableByteChannel open(GcsPath path) throws IOException {
     return googleCloudStorage.open(new StorageResourceId(path.getBucket(), path.getObject()));
   }
 
+  /**
+   * Opens an object in GCS.
+   *
+   * <p>Returns a SeekableByteChannel that provides access to data in the bucket.
+   *
+   * @param path the GCS filename to read from
+   * @param readOptions Fine-grained options for behaviors of retries, buffering, etc.
+   * @return a SeekableByteChannel that can read the object data
+   */
+  @VisibleForTesting
+  SeekableByteChannel open(GcsPath path, GoogleCloudStorageReadOptions readOptions)
```

Comment:
Where is this used ?

Reply:
This is used for the test. Previously test access the implementation detail (e.g. `GoogleCloudStorageReadChannel`) in `testGCSChannelCloseIdempotent()` but it doesn't need to do it anymore with this function.